app/components/support_interface/reference_with_feedback_component.rb

```diff
@@ -215,6 +227,13 @@ def sign_in_as_referee_row
       end
     end
 
+    def confidentiality_row
+      {
+        key: 'Can this reference be shared with the candidate?',
+        value: confidentiality_value,
+      }
+    end
```

Comment:
```suggestion
    def confidentiality_row
      return unless reference.feedback_provided?
      
      {
        key: 'Can this reference be shared with the candidate?',
        value: confidentiality_value,
      }
    end
```

Reply:
This doesn't quite work.

Using return here introduces `nil` into the rows array, which causes an error when the code tries to  access the key (e.g. `row[:key]`). 

We could fix this by compacting the array but I believe it is cleaner and more elegant to conditionally insert the `confidentiality_row` only when it's needed.

Reply:
It looks like the array is already being comapct 🧐

Reply:
The `base_rows` array is being flattened and compacted, yes in the `ReferenceWithFeedbackComponent` but the `ReferenceWithFeedbackComponent` renders the `SummaryCardComponent` within it which processes the `rows` array using `rows_including_actions_if_editable`. This method uses `tap`, which causes an error if the value is `nil`. 

I'd prefer not to update the method in the `SummaryCardComponent` for this as it's referenced 43 times in the codebase.

Reply:
If we  add it to the base rows in the same way as the other components and then compact it, the nil will disappear. Which is the same result as not inserting it into the array in the first place. I think if it's possible to no introduce another pattern, we might as well do that.

Reply:
Is this what you mean? https://github.com/DFE-Digital/apply-for-teacher-training/pull/10137/commits/b2bce5d5051b6470e92c1c3bc7dab6bdb494d418

Sorry both, i'm not sure what rabbit hole I was down...